src/prelude/internals.mo

```diff
@@ -370,7 +370,7 @@ func @new_async<T <: Any>() : (@Async<T>, @Cont<T>, @Cont<Error>) {
 module @ManagementCanister = {
   public type wasm_module = Blob;
   public type canister_settings = {
-    controller : ?Principal;
+    controllers : ?[Principal];
```

Comment:
Why not model `null` with `[]`?

Reply:
Oh, this is an interface given by the IC?

Reply:
Yep, and I expect candid would default to null this way if the field is omitted. Our code was actually out-of-date in a benign way.

Reply:
The empty set of controllers is a very different setting than `null`, which says “use the default ”, which is the caller.